base/io.jl

```diff
@@ -448,7 +456,32 @@ function readuntil(s::IO, t::AbstractString)
 end
 
 readline() = readline(STDIN)
-readline(s::IO) = readuntil(s, '\n')
+
+function readline(s::IO, chomp = false; nl2lf=false)
```

Comment:
Ah, indeed `\r\n` is probably hard to handle efficiently from `readuntil`, so `readline` probably still deserves a custom implementation. That would also allow for an efficient version working on bytes for standard UTF-8  `String`.